File: sdk/go/examples/Brazierfile.go
//go:build ignore

package main

import brazier "github.com/brazier/sdk/go"

func main() {
	p := brazier.NewPipeline(append(
		[]brazier.PipelineOption{
			brazier.UseWorkflow("build-test-deploy", "v1.2.0"),
		},
		brazier.Nodes(
			brazier.Job("lint", brazier.JobSpec{
				Commands: []string{"go vet ./...", "golangci-lint run"},
			}),

			brazier.Stage("test",
				brazier.Job("unit", brazier.JobSpec{
					Commands: []string{"go test ./..."},
				}),
				brazier.Job("integration", brazier.JobSpec{
					Commands: []string{"go test -tags=integration ./..."},
				}),
			).DependsOn("lint"),

			brazier.Job("build", brazier.JobSpec{
				Commands:      []string{"go build -o bin/app ./cmd/app"},
				ArtifactPaths: []string{"bin/app"},
			}).DependsOn("test"),

			brazier.Job("deploy", brazier.JobSpec{
				Commands: []string{"./scripts/deploy.sh"},
				Secrets:  []string{"DEPLOY_TOKEN"},
			}).DependsOn("build").When(brazier.OnBranch("main")),
		)...,
	)...)

	brazier.Run(p)
}

File: sdk/ts/examples/Brazierfile.ts
import { pipeline, useWorkflow, job, stage, run, onBranch } from "../src";

const p = pipeline(
  useWorkflow("build-test-deploy", "v1.2.0"),

  job("lint", { commands: ["eslint .", "tsc --noEmit"] }),

  stage("test",
    job("unit", { commands: ["jest --testPathPattern=unit"] }),
    job("e2e",  { commands: ["playwright test"] }),
  ).dependsOn("lint"),

  job("build", {
    commands: ["npm run build"],
    artifactPaths: ["dist/"],
  }).dependsOn("test"),

  job("deploy", {
    commands: ["./scripts/deploy.sh"],
    secrets: ["DEPLOY_TOKEN"],
  }).dependsOn("build").when(onBranch("main")),
);

run(p);
